src/librustc_back/svh.rs

```diff
@@ -51,19 +51,26 @@ use std::hash::{Hash, SipHasher, Hasher};
 use syntax::ast;
 use syntax::visit;
 
-#[derive(Clone, PartialEq, Debug)]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct Svh {
-    hash: String,
+    raw: u64,
 }
 
 impl Svh {
     pub fn new(hash: &str) -> Svh {
         assert!(hash.len() == 16);
-        Svh { hash: hash.to_string() }
-    }
+        // Ideally we'd just reverse the nibbles on LE machines during to_string, unfortunately
+        // this would break the abi so I guess we're just doing this now.
+
+        let s = if cfg!(target_endian = "big") {
+            hash.to_string()
+        } else {
+            hash.chars().rev().collect()
+        };
```

Comment:
We don't currently maintain ABI compatibility today, so that's not too much of a worry, and I feel like if we cross endianness platforms that this is likely to be the least of our problems (I'm also not 100% sure that this is a problem). Altogether I think it's fine to omit this.